### DISCUSSION TITLE
Custom functionality before result (REST)

### DISCUSSION THREAD

**Discussion by MichalNemec** [supabase/supabase#6920] [General]
Hello,
Im thinking about using supabase, but one thing that baffles me is the customization of rest.

my app relies on language and translations.
The premise is:
If table has column translatable and set to true and the user has different language than en -> call google translate, then return data.

how this could be done in supabase

Supabase ticks a lot of boxes, but im stuck on this one.

i dont mind if i have to modify the core of supabase in order to get this to work.

**Comment by steve-chavez** [comment]
Hello,

You have [edge functions](https://supabase.com/blog/2022/03/31/supabase-edge-functions) that can do anything like calling an endpoint for google translate and combining that with a call to the REST API and then returning the data to the client.

> If table has column translatable and set to true and the user has different language than en -> call google translate, then return data.

That being said, not sure why you would need the translate call to be done server side. You could fetch the REST API data client-side, then do the translate call client-side too.

**Reply by MichalNemec** [reply]
Thank you, ill look at it. I need as less manipulation and data manipulation in the app as possible.

**Reply by MichalNemec** [reply]
i tried it, it seems like i have to send call to functions , so i dont have to use the api in any way.. i was searching a bit and found reedelk for kong, that could work as a wrapper, but im not sure, this is uncharted territory for me. Last week i tried directus, created hook and it was done, but im not 100% sure about using it, because i will use mainly flutter and angular and i like the libraries for communication from supabase.

**Reply by steve-chavez** [reply]
Another option might be using https://github.com/pramsey/pgsql-http, so you can do a direct call to google translate inside SQL.
